File: 0979-di-string-match/0979-di-string-match.java
class Solution {
    public int[] diStringMatch(String s) {
        int start=0;
        int result[]=new int[s.length()+1];

        
        int end=result.length-1;
        int index=0;

        while(index<s.length())
        {
            if(s.charAt(index)=='I')
            {
                result[index]=start;
                start++;
            }else
            {
                result[index]=end;
                end--;
            }
            index++;
        }

        result[index]=end;
        return result;
        
    }
}

File: 0979-di-string-match/0979-di-string-match.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        ei=0
        d=len(s)
        res=[]

        for i in range(0,len(s)):
            if s[i]=='I':
                res.append(ei)
                ei+=1
                
            else:
                res.append(d)
                d-=1
        res.append(d)
        return res

        